test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -167,6 +167,27 @@
                 (is (= 403 (:status response))
                     (ks/pprint-to-string response))))))))))
 
+(deftest ^:integration certificate-with-ca-true-extension-refused
+  (testing (str "Validates that the server rejects a csr for signing"
+                " that has the v3 CA:TRUE extension")
+    (let [master-conf-dir (str test-resources-dir 
+                               "/ca_true_test/master/conf")]
+      (bootstrap/with-puppetserver-running
+        app
+        {:jruby-puppet {:master-conf-dir master-conf-dir}}
+        (let [response (http-client/put
+                       (str "https://localhost:8140"
+                            "puppet-ca/v1/certificate_status/test_cert_ca_true")
+                       {:ssl-cert (str master-conf-dir "/ssl/ca/ca_crt.pem")
+                        :ssl-key (str master-conf-dir "/ssl/ca/ca_key.pem")
+                        :ssl-ca-cert (str master-conf-dir "/ssl/ca/ca_crt.pem")
+                        :as :text
+                        :body "{\"desired_state\": \"signed\"}"
+                        :headers {"content-type" "application/json"}})]
+        (is (= 409 (:status response)))
+        (is (.startsWith (:body response) "Found extensions"))
+        (is (.contains (:body response) "2.5.29.19")))))))
```

Comment:
nitpick and I'm not sure any changes are required here, but...

This test is written to assume that the test_cert_ca_true CSR pem file has that extension set in it; it doesn't actually verify that before making the request.  It might be nice to have an assertion before any of the HTTP stuff that verifies that that pem actually contains that extension.

That said, I can't think of anything that would go wrong without being detected with the test as-is; e.g. if someone replaced the fixture CSR pem with another pem that didn't have that extension in it, presumably this test would fail because either we'd get a different status code, or the body wouldn't contain that OID.  So I think it's probably not necessary to add the assertion that I suggested.

One other minor comment: it seems like there are a handful of pems in the fixture directory that aren't actually used by or necessary for this test.  e.g. `revoked-agent.pem` - actually that might be the only one.

It might spare some confusion to remove that file if it's not actually needed.

That said, I'd be OK merging this as is... +1

so @erikPrime let us know if you think either of those comments are worth addressing; if not, we can go ahead and push the button.